src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/PlainPackageFeedBase.cs

```diff
@@ -43,5 +44,16 @@ public Task<SearchResult<IPackageSearchMetadata>> SearchAsync(string searchText,
 
         public Task<SearchResult<IPackageSearchMetadata>> RefreshSearchAsync(RefreshToken refreshToken, CancellationToken cancellationToken)
             => Task.FromResult(SearchResult.Empty<IPackageSearchMetadata>());
+
+        internal static FeedSearchContinuationToken ThrowIfNotFeedSearchContinuationToken(ContinuationToken continuationToken)
+        {
+            var searchToken = continuationToken as FeedSearchContinuationToken;
+            if (searchToken == null)
+            {
+                throw new InvalidOperationException(Strings.Exception_InvalidContinuationToken);
+            }
+
+            return searchToken;
```

Comment:
We don't need a method for this. Instead, we can write this as follows in-line wherever it's used:

```suggestion
var searchToken = continuationToken as FeedSearchContinuationToken ?? throw new InvalidOperationException(Strings.Exception_InvalidContinuationToken);
```